draft-ietf-mls-protocol.md

```diff
@@ -1435,48 +1461,220 @@ all arrive at the following state:
 
 # Message Protection
 
-[[ OPEN ISSUE: This section has initial considerations about message
-protection.  This issue clearly needs more specific recommendations,
-possibly a protocol specification in this document or a separate
-one. ]]
+The primary purpose of the handshake protocol is to provide an authenticated
+group key exchange to participants. In order to protect Application messages
+sent among those participants, the Application secret provided by the Handshake
+key schedule is used to derive encryption keys for the Message Protection Layer.
+
+Application messages MUST be protected with the Authenticated-Encryption
+with Associated-Data (AEAD) encryption scheme associated with the MLS ciphersuite.
+Note that "Authenticated" in this context does not mean messages are known to
+be sent by a specific participant but only from a legitimate member of the group.
+To authenticate a message from a particular member, signatures are required.
+Handshake messages MUST use asymmetric signatures to strongly authenticate
+the sender of a message.
+
+Each participant maintains their own chain of Application secrets, where the first
+one is derived based on a secret chained from the Epoch secret.
+As shown in {{key-schedule}}, the initial Application secret is bound to the
+identity of each participant to avoid collisions and allow support for decryption
+of reordered messages.
+
+Subsequent Application secrets MUST be rotated for each message sent in
+order to provide stronger cryptographic security guarantees. The Application
+Key Schedule use this rotation to generate fresh AEAD encryption keys and nonces
+used to encrypt and decrypt future Application messages.
+In all cases, a participant MUST NOT encrypt more than expected by the security
+bounds of the AEAD scheme used.
+
+Note that each change to the Group through a Handshake message will cause
+a change of the Group Secret. Hence this change MUST be applied before encrypting
+any new Application message. This is required for confidentiality reasons
+in order for Members to avoid receiving messages from the group after leaving,
+being added to, or excluded from the Group.
+
+## Application Key Schedule {#key-schedule-application}
+
+After computing the initial Application Secret shared by the group,
+each Participant creates an initial Participant Application Secret
+to be used for its own sending chain:
 
-The primary purpose of this protocol is to enable an authenticated
-group key exchange among participants.  In order to protect messages sent among
-those participants, an application will need to specify how messages are protected.
+~~~
+           application_secret_[0]
+                     |
+                     V
+           Derive-Secret(., "app upd", [sender])
+                     |
+                     V
+           application_secret_[1]
```

Comment:
Thanks, this is clearer.  Two minor edits / quetions:

However, the subscripts should indicate that this is a fanout.  I would propose having the root be something like `application_secret` (no subscript), then have the specific application secrets be double-subscripted, e.g., `app_secret_[sender]_[0]`.

You might want to use a separate label (app sender?) for this stage as opposed to the linear updates.

Reply:
Fixed, Thanks ! : )